File: Week2 Assignment/SparseMatrix.py
#Check if matrix is sparse

rows = int(input("Enter number of rows: "))
cols = int(input("Enter number of columns: "))

print("Enter the matrix row-wise:")
matrix = [list(map(int, input().split())) for _ in range(rows)]

zero_count = sum(cell == 0 for row in matrix for cell in row)
total_elements = rows * cols

if zero_count > total_elements // 2:
    print("The matrix is a sparse matrix.")
else:
    print("The matrix is NOT a sparse matrix.")

File: Week2 Assignment/SparseMatrix.c
//Check if matrix is sparse

#include <stdio.h>

int main() {
    int rows, cols, i, j;
    int count = 0;

    printf("Enter number of rows and columns: ");
    scanf("%d %d", &rows, &cols);

    int matrix[rows][cols];

    printf("Enter matrix elements:\n");
    for (i = 0; i < rows; i++) {
        for (j = 0; j < cols; j++) {
            scanf("%d", &matrix[i][j]);
            if (matrix[i][j] == 0)
                count++;
        }
    }

    int total = rows * cols;

    if (count > total / 2)
        printf("The matrix is a sparse matrix.\n");
    else
        printf("The matrix is NOT a sparse matrix.\n");

    return 0;
}

